### PR TITLE
bump to kubernetes-1.17.0-beta.2

### PR DESCRIPTION
**Pull request by mfojtik**
This moves openshift-apiserver to kubernetes 1.17.0-beta.2 which include dynamic certificate reloading and other fixes in aggregated apiservers.

The apiserver-library-go was also bumped to 1.17.0-beta.2: https://github.com/openshift/apiserver-library-go/tree/openshift-apiserver-4.3-kubernetes-1.17.0-beta.2 (surprisingly tolerating library-go still being on 1.16.2).

### CODE REVIEW

**Comment by mfojtik** [comment]
@deads2k as promised

**Comment by mfojtik** [comment]
/hold

we need kube tag to move forward

**Comment by mfojtik** [comment]
@sttts the openapi generation fail with: 

```
E1119 14:01:39.582048   59789 openapi.go:304] Error when generating: managementState, ManagementState github.com/openshift/api/operator/v1.ManagementState
2019/11/19 14:01:39 OpenAPI code generation error: Failed executing generator: some packages had errors:
Tags in comment and struct should match for member (ManagementState) of (github.com/openshift/api/samples/v1.ConfigStatus)
```

/cc @damemi

File: pkg/apps/apis/apps/log_options_test.go [hunk 1]
```diff
@@ -15,6 +15,10 @@ func TestLogOptionsDrift(t *testing.T) {
 		// Verify name
 		name := popts.Field(i).Name
 		doptsField, found := dopts.FieldByName(name)
+		// TODO: Add this option to deployment config log options
```
Comment (tnozicka) [inline pkg/apps/apis/apps/log_options_test.go]: followup with a BZ seems fine to me

File: pkg/cmd/openshift-apiserver/server.go [hunk 2]
```diff
@@ -11,8 +11,8 @@ import (
 
 	openshiftcontrolplanev1 "github.com/openshift/api/openshiftcontrolplane/v1"
 	"github.com/openshift/openshift-apiserver/pkg/cmd/openshift-apiserver/openshiftapiserver"
-	_ "k8s.io/kubernetes/pkg/client/metrics/prometheus" // for client metric registration
-	_ "k8s.io/kubernetes/pkg/util/workqueue/prometheus" // for workqueue metric registration
+	// _ "k8s.io/kubernetes/pkg/client/metrics/prometheus" // for client metric registration
```
Comment (deads2k) [inline pkg/cmd/openshift-apiserver/server.go]: this moved to component-base.  Please update similarly

Comment (deads2k) [reply pkg/cmd/openshift-apiserver/server.go]: component-base/metrics/prometheus/restclient

File: pkg/authorization/apiserver/registry/subjectaccessreview/rest.go [hunk 3]
```diff
@@ -151,7 +151,7 @@ func (r *REST) isAllowed(user user.Info, sar *authorizationapi.SubjectAccessRevi
 		}
 	}
 
-	authorized, reason, err := r.authorizer.Authorize(localSARAttributes)
+	authorized, reason, err := r.authorizer.Authorize(context.TODO(), localSARAttributes)
```
Comment (deads2k) [inline pkg/authorization/apiserver/registry/subjectaccessreview/rest.go]: this should pass through ctx.

File: pkg/apps/apis/apps/log_options_test.go (hunk 1 above)
Comment (deads2k) [reply pkg/apps/apis/apps/log_options_test.go]: > followup with a BZ seems fine to me

Yeah, BZ for 4.4.  it's not urgent.

File: pkg/authorization/apis/authorization/v1/zz_generated.conversion.go [hunk 4]
```diff
@@ -226,16 +176,6 @@ func RegisterConversions(s *runtime.Scheme) error {
 	}); err != nil {
 		return err
 	}
-	if err := s.AddGeneratedConversionFunc((*v1.SelfSubjectRulesReviewSpec)(nil), (*authorization.SelfSubjectRulesReviewSpec)(nil), func(a, b interface{}, scope conversion.Scope) error {
```
Comment (deads2k) [inline pkg/authorization/apis/authorization/v1/zz_generated.conversion.go]: do we still have fuzz tests for the API in this repo?

File: pkg/authorization/apiserver/registry/subjectaccessreview/rest.go (hunk 3 above)
Comment (deads2k) [inline pkg/authorization/apiserver/registry/subjectaccessreview/rest.go]: should be ctx, not todo

**Comment by deads2k** [comment]
fairly minor comments.  The biggest one is the question of fuzzing tests.

File: pkg/authorization/apiserver/registry/subjectaccessreview/rest.go (hunk 3 above)
Comment (mfojtik) [reply pkg/authorization/apiserver/registry/subjectaccessreview/rest.go]: hups

File: pkg/cmd/openshift-apiserver/server.go (hunk 2 above)
Comment (mfojtik) [reply pkg/cmd/openshift-apiserver/server.go]: will do, thanks

File: pkg/authorization/apis/authorization/v1/zz_generated.conversion.go (hunk 4 above)
Comment (mfojtik) [reply pkg/authorization/apis/authorization/v1/zz_generated.conversion.go]: @deads2k we have some fuzz tests in pkg/api/serialization_test.go, I don't think we have more tests.

**Comment by mfojtik** [comment]
Follow up list:

1) Make a BZ for new log options field
2) Make a BZ doe @openshift/openshift-team-developer-experience for disabled unit test

**Comment by mfojtik** [comment]
@deads2k whoah e2e passed :)

**Comment by mfojtik** [comment]
@deads2k @sttts @adambkaplan @bparees this is ready for review

File: pkg/apps/apis/apps/v1/zz_generated.conversion.go [hunk 5]
```diff
@@ -174,16 +174,6 @@ func RegisterConversions(s *runtime.Scheme) error {
 	}); err != nil {
 		return err
 	}
-	if err := s.AddGeneratedConversionFunc((*v1.DeploymentTriggerImageChangeParams)(nil), (*apps.DeploymentTriggerImageChangeParams)(nil), func(a, b interface{}, scope conversion.Scope) error {
```
Comment (deads2k) [inline pkg/apps/apis/apps/v1/zz_generated.conversion.go]: reminder for future me.  These were actually registered twice before.  The fuzz tests appear to be working, so we're good.

**Pull request merged**